operator/v1/types_dns.go

```diff
@@ -101,6 +100,39 @@ type DNSSpec struct {
 	// +optional
 	// +kubebuilder:default=Normal
 	LogLevel DNSLogLevel `json:"logLevel,omitempty"`
+
+	// cache describes the caching configuration that applies to all server blocks listed in the Corefile.
+	// This field allows a cluster admin to optionally configure:
+	// * successTTL which is a duration for which positive responses should be cached.
+	// * denialTTL which is a duration for which negative responses should be cached.
+	// If this is not configured, OpenShift will configure success and denial caching with a default value that is
+	// subject to change. At the time of writing, the default successTTL is 900 seconds and the default denialTTL is 30
+	// seconds or as noted in the respective Corefile for your version of OpenShift.
+	// +optional
+	Cache DNSCache `json:"cache,omitempty"`
+}
+
+// DNSCache defines the fields for configuring DNS caching.
+type DNSCache struct {
+	// successTTL is optional and specifies the amount of time that a positive response should be cached.
+	//
+	// If configured, it must be a value of 1s (1 second) or greater up to a theoretical maximum of several years.
+	// If not configured, a default value of 900 seconds will be used unless noted otherwise in the respective
+	// Corefile for your version of OpenShift.
+	// +kubebuilder:validation:Pattern=^(([1-9][0-9]*(\.[0-9]+)?(s|m|h|d))+)$
+	// +kubebuilder:validation:Type:=string
+	// +optional
+	SuccessTTL *metav1.Duration `json:"successTTL,omitempty"`
```

Comment:
Is a zero value valid here? What would happen if I put this value as `0s`?

Looking at the validation, that doesn't appear to be a valid option, in which case, I think this doesn't need to be a pointer, let's make it a reference instead and your controller can default it when it sees a 0 value

Reply:
Fixed in the latest push!

Reply:
Fwiw `0s` isn't valid with the regex, but `0` is.  I think that was the intention though.

Reply:
Reading through https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required I'm inclined to push back to pointers, since even though we don't care between 0 and nil and for both we can easily treat them as unset. The added benefit of pointer is the implicit nature of optional.

Reply:
This is a configuration API right? I've always been told that for configuration APIs we should avoid pointers as they harm discoverability for end users. If they can see the option is there (because something marshalled it) then they might have a chance of understanding that they can actually set it/discover new features they hadn't previously been aware of

Reply:
@JoelSpeed @soltysh what's the final verdict on this? Pointers or no pointers?

Reply:
Let's leave it as is.